torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -671,8 +662,16 @@ def view_avoid_dupes_with_primals(t):
         # This analysis function returns *only* the outputs that are meant to be tangents to the backwards.
         # Anything that aliases (inputs returned in the fw due to metadata mutations, or outputs that alias inputs/intermediates)
         # are *regenerated* later, and not used directly in the autograd graph
+        def _plain_fake_tensor_like_subclass(x):
+            t = torch.empty(x.shape, dtype=x.dtype, device=x.device)
+            return FakeTensor.from_tensor(t, detect_fake_mode())
+
         f_input_tangents = [
-            inp
+            # For subclass inputs that are mutated out of graph tangent will be
+            # plain zeros tensor like subclass input.
```

Comment:
hmm I think this comment doesn't really give the full picture. I would probably say something like:
```
# Generally when creating tangents to trace with, we assume that tangents will have
# the same subclass-ness as their forward outs
# however: for tangents that correspond to input mutations, in practice it is more likely
# that these tangents will be plain tensors of zeros at runtime, so we tweak our guess
# to assume that the these tangents should always be plaint tensors.
# Example:
#  def f(x):
        x.mul_(2)
        return x + 1
#  out = f(x)
#  out.sum().backward()
# In the above code, we will have a tangent "x_updated_tangent",
# which will be a plain tensor of zeros, *unless* x is used in some compute after executing f
```